examples/client/Locomotion/src/pages/Payments/credit-cards.tsx

```diff
@@ -1,27 +1,57 @@
 import React, { useEffect, useState } from 'react';
 import { View } from 'react-native';
+import { HeaderLink } from '../../Components/Menu/styled';
 import { MAIN_ROUTES } from '../routes';
 import i18n from '../../I18n';
 import {
-  DeleteCreditCard,
-  DeleteCreditCardText,
   CardsListContainer,
   PaymentMethodsContainer,
-  CreditCardsContainer,
+  MethodCard,
+  ChevronIcon,
+  ChangeButton,
 } from './styled';
+import {
+  CardContainer, CardContantContainer, CardTitle, CardTitleContainer,
+} from '../../Components/InformationCard/styled';
 
 import PaymentMethod from '../../Components/CardRow';
 import PaymentsContext from '../../context/payments';
-import cashPaymentMethod from './cashPaymentMethod';
 import { navigate } from '../../services/navigation';
+import ChoosePaymentMethod from '../../popups/ChoosePaymentMethod';
+import cashPaymentMethod from './cashPaymentMethod';
+import chevronIcon from '../../assets/chevron.svg';
 
 export default ({
-  onDetach = (id: string) => null,
   loadingState = false,
   onAddClick = undefined,
 }) => {
-  const [loading, setLoading] = useState(false);
   const usePayments = PaymentsContext.useContainer();
+  const [loading, setLoading] = useState(false);
+  const [defaultMethod, setDefaultMethod] = useState({ ...usePayments.getClientDefaultMethod(), mark: true });
+  const [showChoosePayment, setShowChoosePayment] = useState(false);
+
+  const onSubmit = async (payment : any) => {
+    const chosenDefault = usePayments.paymentMethods.find(({ id }) => id === payment) || defaultMethod;
+    if (chosenDefault === defaultMethod) {
+      return;
+    }
+
+    await usePayments.updatePaymentMethod(defaultMethod?.id, { isDefault: false });
```

Comment:
1. u can use promise.all
2. use the same network/hooks in the backend, that doesn't make sense to create 2 call, what if one of them fails?

Reply:
moving it to payment ms

Reply:
Sababa, u can use async hooks